protocol.md

```diff
@@ -568,6 +568,127 @@ Tus-Resumable: 1.0.0
 Upload-Offset: 11
 ```
 
+### Partial Checksum
+
+This extension allows Clients that use the Checksum extension to verify and confirm the
+integrity of chunks that were not received in full by the Server - thereby avoiding that
+intact data needs to be sent again.
```

Comment:
The Partial Checksum extension builds on top of the Checksum extension, which stays intact for completed requests.

In your example the `PATCH` request for bytes 50-100 would - to satisfy the Checksum extension requirements - already include an `Upload-Checksum` header with the checksum of those 50 bytes.

Per the Checksum extension, the Server would check if the checksum for those 50 bytes would match what the Client sent in the `Upload-Checksum` header. If it does not, the Server would respond with a `460 Checksum Mismatch` error and discard the 50 bytes. Which I'd argue is the right thing to do when the Checksum for the fully received data doesn't match: the Client should still come up with the same checksum as the one it sent originally - and if it is tolerant to checksum errors, it could probably skip using the Checksum and Partial Checksum extensions altogether.

From the server-side, I'd think differentiating between partial/interrupted and fully received requests should be relatively simple, as the server can assume a partial/interrupted request when the received request body's byte count is less than the number in the request's `Content-Length` header - and a fully received request if byte count and header value match.

Reply:
I do now how the checksum extension work. ;) My comment was more to highlight a different angle on how it could be implemented. As long as it's clear how this extension behaves when both this and checksum is supported I don't think there is an issue here. The use case proposed (for invalid checksum) is very niche and just an idea.